utils/notificationhubServicePartition.ts

```diff
@@ -26,19 +28,45 @@ export function getNHLegacyConfig(envConfig: IConfig): NotificationHubConfig {
 }
 
 /**
- * It returns an ExtendedNotificationHubService related to one of the new Notification Hub instances
+ *
+ * @param sha The sha to test
+ * @returns
+ */
+export const testShaForPartitionRegex = (regex: string, sha: InstallationId) =>
+  new RegExp(regex).test(sha);
+/**
+ * @returns A connection string based on NH namespace and sharedAccessKey provided
+ */
+const buildNHConnectionString = ({
+  namespace,
+  sharedAccessKey
+}: NotificationHubPartitions): string =>
+  `Endpoint=sb://${namespace}.servicebus.windows.net/;SharedAccessKeyName=DefaultFullSharedAccessSignature;SharedAccessKey=${sharedAccessKey}`;
+
+/**
+ * It returns the configuration related to one of the new Notification Hub instances
  * based on the partion mechanism defined
- * @param fiscalCodeHash a valid hash256 representing a Fiscal Code
+ * @param envConfig the env config with Notification Hub connection strings and names
+ * @param sha a valid hash256 representing a Fiscal Code
  */
-export function getNHService(
-  fiscalCodeHash: string
-): ExtendedNotificationHubService {
-  // tslint:disable-next-line: no-tslint-disable-all
-  // tslint:disable-next-line
-  const fs = fiscalCodeHash;
-
-  throw new Error("It should not be called");
-}
+export const getNotificationHubPartitionConfig = (envConfig: IConfig) => (
+  sha: InstallationId
+): NotificationHubConfig => {
+  const partition = envConfig.AZURE_NOTIFICATION_HUB_PARTITIONS.find(p =>
+    testShaForPartitionRegex(p.partitionRegex, sha)
+  );
+
+  if (partition) {
+    const connectionString = buildNHConnectionString(partition);
+
+    return {
+      AZURE_NH_ENDPOINT: connectionString as NonEmptyString,
+      AZURE_NH_HUB_NAME: partition.name
+    };
+  }
+
+  throw new Error(`Unable to find Notification Hub partition for ${sha}`);
+};
```

Comment:
Retrieve the partition whose regex matches the installationId and return it as `NotificationHubConfig`